src/HomePage/HomePage.tsx

```diff
@@ -1,5 +1,75 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { StyledImage } from "../StyledApp";
+import axios from "axios";
+import { StyledDivWrapper } from "./StyledHomepage";
+import Loader from "../styles/StyledLoader";
 
-export default function Home_page() {
-  return <div>Home Page</div>;
+interface Painting {
+  objectID: number;
+  accessionYear: number;
+  primaryImage: string;
+  artistDisplayName: string;
+  title: string;
 }
+
+const HomePage: React.FC = () => {
+  // const [uniqueIdList, setUniqueIdList] = useState<Array<number>>([]);
+  // const [randomID, setRandomID] = useState<number | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [paintingData, setPaintingData] = useState<Painting>({
+    objectID: 0,
+    accessionYear: 0,
+    primaryImage: "",
+    artistDisplayName: "",
+    title: "",
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setLoading(true);
+      try {
+        //Fetch object IDs
+        const response = await axios.get(
+          "https://collectionapi.metmuseum.org/public/collection/v1/objects?departmentIds=11"
+        );
+        const uniqueIdList = response.data.objectIDs;
+
+        // Select random ID
+        const randomIndex = Math.floor(Math.random() * uniqueIdList.length);
+        const randomID = uniqueIdList[randomIndex];
+
+        // Fetch painting data using the random ID:
+        const paintingResponse = await axios.get(
+          `https://collectionapi.metmuseum.org/public/collection/v1/objects/${randomID}`
+        );
+        setPaintingData(paintingResponse.data);
+        setLoading(false);
+      } catch (error) {
+        console.error("Error fetching data:", error);
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, []);
+
+  if (loading) {
+    return <Loader></Loader>;
+  }
+
+  return (
+    <StyledDivWrapper>
+      <h1>ID: {paintingData.objectID}</h1>
+      {paintingData && (
+        <>
+          <h2>Title: {paintingData.title}</h2>
+          <h3>Artist: {paintingData.artistDisplayName}</h3>
+        </>
+      )}
+      {paintingData.primaryImage && (
```

Comment:
question: why do we have this condition?

Reply:
This conditions checks whether primaryImage attribute is present in the paintingData object (some of them does not have an image)